chain/epoch-manager/src/reward_calculator.rs

```diff
@@ -13,15 +13,27 @@ use crate::validator_stats::get_validator_online_ratio;
 pub(crate) const NUM_NS_IN_SECOND: u64 = 1_000_000_000;
 pub const NUM_SECONDS_IN_A_YEAR: u64 = 24 * 60 * 60 * 365;
 
+/// Contains online thresholds for validators.
+#[derive(Clone, Debug)]
+pub struct ValidatorOnlineThresholds {
+    /// Online minimum threshold below which validator doesn't receive reward.
+    pub online_min_threshold: Rational32,
+    /// Online maximum threshold above which validator gets full reward.
+    pub online_max_threshold: Rational32,
+    /// If set, contains a number between 0 and 100 (percentage), and endorsement ratio
+    /// below this threshold will be treated 0, and otherwise be treated 1,
+    /// before calculating the average uptime ratio of the validator.
+    /// If not set, endorsement ratio will be used as is.
+    pub endorsement_cutoff_threshold: Option<u8>,
+}
+
 #[derive(Clone, Debug)]
 pub struct RewardCalculator {
     pub max_inflation_rate: Rational32,
     pub num_blocks_per_year: u64,
     pub epoch_length: u64,
     pub protocol_reward_rate: Rational32,
     pub protocol_treasury_account: AccountId,
-    pub online_min_threshold: Rational32,
```

Comment:
move these to the `ValidatorOnlineThresholds` struct passed to `calculate_reward`.
In fact we do not re-initialize the `RewardCalculator` when there is a new epoch or protocol change, so this way it is safer as we will need to change the online thresholds (endorsement cutoff in this case) via protocol upgrades.